### PR TITLE
OCPBUGS-16036: Set status on CR properly when STS provisioned

### PR DESCRIPTION
**Pull request by bentito**

### CODE REVIEW

File: pkg/operator/credentialsrequest/credentialsrequest_controller.go [hunk 1]
```diff
@@ -1040,3 +1048,36 @@ func checkForFailureConditions(cr *minterv1.CredentialsRequest) bool {
...
+			// Conflict error means that the object has been modified by another process
+			// Get the latest version of the object and retry the update operation
+			err := r.Client.Get(context.Background(), types.NamespacedName{Name: cr.Name, Namespace: cr.Namespace}, cr)
+			if err != nil {
+				return err
+			}
+			return r.UpdateProvisionedStatus(cr, provisioned)
+		}
+		return err
+	}
```
Comment (fxierh) [inline pkg/operator/credentialsrequest/credentialsrequest_controller.go:1081]: I'm wondering if we could use retry.RetryOnConflict() here

Comment (bentito) [reply pkg/operator/credentialsrequest/credentialsrequest_controller.go]: Done in [4f25b93](https://github.com/openshift/cloud-credential-operator/pull/562/commits/4f25b9368c7c03d1ea90f469ced8162dd70e6653) Thanks Feilian, yeah, no sense reinventing the wheel!

**Pull request merged**